### DISCUSSION TITLE
External sensor driver as a module

### DISCUSSION THREAD

**Discussion by nobodyguy** [zephyrproject-rtos/zephyr#46098] [Q&A] [answered]
Hello, has anyone got external sensor driver working as a west module?
I have T2 project tree structure shown here - https://gist.github.com/nobodyguy/fcf7b32e8ee1300d72ea454dd005dff9
As you can also see, I added module dependency on abc123 module to **my_firmware/west.yml**
For structure of abc123 module (with single sensor driver with the same name), I followed this example - https://github.com/iwasz/example-zephyr-module
I build **my_firmware** from within top level **/my_workspace** with this command: `west build -p -b nrf52832_custom_board -d build/my_firmware -s my_firmware`
And the result is this: `/my_workspace/my_firmware/prj.conf:95: warning: attempt to assign the value 'y' to the undefined symbol ABC123`

**Comment by nobodyguy** [comment, marked as answer]
Resolved - my setup was correct, but west kept resetting the git HEAD back to the latest commit and since I was just testing this concept, all the important CMake and Kconfig changes were uncommited so it didn't work as expected.

**Reply by yashi** [reply]
When `west build` runs, I'm pretty sure that it does NOT reset the heads.  Otherwise, I can't build with my test code while developing.  Are you sure about the reset?

**Reply by nobodyguy** [reply]
`west update`

**Reply by yashi** [reply]
That's the command to reset your module back to the latest commit.
https://docs.zephyrproject.org/3.0.0/guides/west/built-in.html#west-update

You don't need to `west update` every time you build your software.

**Reply by nobodyguy** [reply]
That's right, but when you're tinkering with moving custom drivers to external modules, you use `west update` a lot and the conseqences may not be so obvious. I was just too focused on my CMake or Kconfig files and thought that they are somehow wrong ;-)
